internal/envoy/run.go

```diff
@@ -78,15 +78,21 @@ func (r *Runtime) Run(ctx context.Context, args []string) (err error) {
 
 	// Block until we receive SIGINT or are canceled because Envoy has died
 	<-sigCtx.Done()
+
 	if cmd.ProcessState != nil && !r.opts.DontArchiveRunDir {
 		return r.archiveRunDir()
 	}
 
 	r.handleShutdown(ctx)
 
-	// Block until the process is complete. This ensures file descriptors are closed.
-	<-waitCtx.Done()
-
+	// At this point, shutdown hooks have run and Envoy is interrupted.
+	// Block until it exits to ensure file descriptors are closed prior to archival.
+	// Allow up to 5 seconds for a clean stop, killing if it can't for any reason.
+	select {
+	case <-waitCtx.Done():
+	case <-time.After(5 * time.Second):
```

Comment:
so the key to unblocking windows is to allow graceful shutdown, but nuke it like this. I used 5 seconds because I haven't lost my thumb